### PR TITLE
Add Rise integration

### PR DESCRIPTION
**Pull request by webmaster128**
This is based on  #417 as discussed with @vekexasia and includes all of today's API changes.

- [x] Rename all symbols with acronyms
- [x] Remove Lisk nethash from package
- [x] Rename "liskCodec" in test description
- [x] Find reference for RISE derivation paths
- [x] Use original RISE transaction in "can parse transaction"
- [x] Don't wait for block after posting a transaction to the network
- [x] Update README after: Rename IovWriter to MultiChainSigner #416

### CODE REVIEW

File: packages/iov-rise/src/risecodec.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  /**
+   * UTF-8 encoded JSON that can be posted to
+   * https://app.swaggerhub.com/apis/LiskHQ/Lisk/1.0.30#/Transactions/postTransaction
+   */
+  bytesToPost: (signed: SignedTransaction): PostableBytes => {
+    switch (signed.transaction.kind) {
+      case TransactionKind.Send:
+        const timestamp = signed.primarySignature.nonce.toNumber();
+        const liskTimestamp = timestamp - 1464109200;
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/risecodec.ts]: riseTimestamp

File: packages/iov-rise/src/risecodec.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          signed.transaction.amount.fractional,
+        );
+
+        const postableObject = {
+          type: 0,
+          amount: amount.toNumber(),
+          recipientId: signed.transaction.recipient,
+          senderId: pubkeyToAddress(signed.primarySignature.publicKey.data),
+          senderPublicKey: Encoding.toHex(signed.primarySignature.publicKey.data),
+          timestamp: liskTimestamp,
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/risecodec.ts]: riseTimestamp

File: packages/iov-rise/src/risecodec.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  },
+
+  /**
+   * ASCII-encoded address string, e.g. 6076671634347365051L
+   *
+   * Addresses cannot be stored as raw uint64 because there are two types of recipient addresses
+   * on the Lisk blockchain that cannot be encoded as uint64 :
+   * 1. leading zeros make different addresses ("123L" != "000123L")
+   * 2. some addresses exceed the uint64 range (e.g. "19961131544040416558L")
+   * These are bugs we have to deal with.
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/risecodec.ts]: Switch to rise example

File: packages/iov-rise/src/serialization.spec.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+import { Encoding } from "@iov/encoding";
+import { Algorithm, ChainId, PublicKeyBytes, SignatureBytes } from "@iov/tendermint-types";
+
+import { amountFromComponents, serializeTransaction, toRiseTimestamp, transactionId } from "./serialization";
+
+const { fromAscii, fromHex } = Encoding;
+
+// use nethash as chain ID
+const riseTestnet = "e90d39ac200c495b97deb6d9700745177c7fc4aa80a404108ec820cbeced054c" as ChainId;
+const liskEpochAsUnixTimestamp = 1464109200;
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/serialization.spec.ts]: some lisk references, like liskEpochAsUnixTimestamp, test descriptions

File: packages/iov-rise/types/parse.d.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    static riseAmount(str: string): AmountFields;
+    static fromRiseTimestamp(riseTimestamp: number): ReadonlyDate;
+    /**
+     * Convert a point in time to a nonce used in the Lisk codec.
+     *
+     * The nonce stores a UNIX timestamp as integer in seconds resolution.
+     * Since we use a Long variable, we are not affected by the year 2038 problem.
+     * The full range of possible Lisk timestamps (epoch in 2016 +/- 68 years)
+     * is covered since we allow negative nonce values.
+     *
```
Comment (LucasIsasmendi) [inline packages/iov-rise/types/parse.d.ts]: Lisk reference on description

File: packages/iov-rise/src/serialization.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import { FullSignature, TransactionIdBytes, TransactionKind, UnsignedTransaction } from "@iov/bcp-types";
+import { Sha256 } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
+
+export function toRiseTimestamp(date: ReadonlyDate): number {
+  const timestamp = Math.floor(date.getTime() / 1000);
+
+  const liskEpoch = Date.UTC(2016, 4, 24, 17, 0, 0, 0) / 1000;
+  const liskTimestamp = timestamp - liskEpoch;
+
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/serialization.ts]: several lisk references on this file

File: packages/iov-rise/src/serialization.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  // Lisk timestamp must be in the signed int32 range (to be stored in a Postgres
+  // integer column and to be serializeable as 4 bytes) but has no further
+  // plausibility restrictions.
+  // https://github.com/LiskHQ/lisk/blob/v1.0.3/logic/transaction.js#L674
```
Comment (LucasIsasmendi) [inline packages/iov-rise/src/serialization.ts]: https://github.com/RiseVision/rise-node/blob/v1.1.1/src/logic/transaction.ts#L346

**Comment by webmaster128** [comment]
:tada: thanks everyone

**Comment by vekexasia** [comment]
:balloon: tkz @webmaster128  and @LucasIsasmendi

**Pull request merged**